cozy/structures/heaps.py

```diff
@@ -131,6 +131,51 @@ def storage_size(self, e : Exp, storage_size) -> Exp:
         assert type(e.type) in (TMinHeap, TMaxHeap)
         return storage_size(EHeapElems(e).with_type(TBag(e.type.elem_type)))
 
+    def maintenance_cost(self,
+            e                   : Exp,
+            solver,
+            op                  : Op,
+            storage_size,
+            maintenance_cost,
+            freebies            : [Exp] = []):
+        assert type(e.type) in (TMinHeap, TMaxHeap)
+
+        from cozy.state_maintenance import mutate
```

Comment:
Rather than doing this import, you could simply have the top-level `maintenance_cost` procedure pass `e_prime` (called `new_value` in this function) as an argument.